ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/services/admin/ApiPackageManagementService.scala

```diff
@@ -74,30 +77,45 @@ class ApiPackageManagementService(
         FutureConverters
           .toScala(res._1)
           .flatMap {
-            case UploadPackagesResult.Ok =>
+            //TODO BH: first step to get this to compile -- need to implement properly polling and retrieval of response event
+            case SubmissionResult.Acknowledged =>
               Future.successful(UploadDarFileResponse())
-            case r @ UploadPackagesResult.Overloaded =>
-              Future.failed(ErrorFactories.resourceExhausted(r.description))
-            case r @ UploadPackagesResult.InternalError(_) =>
+            case r @ SubmissionResult.InternalError(_) =>
               Future.failed(ErrorFactories.internal(r.reason))
-            case r @ UploadPackagesResult.InvalidPackage(_) =>
-              Future.failed(ErrorFactories.invalidArgument(r.description))
-            case r @ UploadPackagesResult.ParticipantNotAuthorized =>
-              Future.failed(ErrorFactories.permissionDenied(r.description))
-            case r @ UploadPackagesResult.NotSupported =>
-              Future.failed(ErrorFactories.unimplemented(r.description))
+            case SubmissionResult.NotSupported =>
+              Future.failed(ErrorFactories.unimplemented("not supported"))
+            case SubmissionResult.Overloaded =>
+              Future.failed(ErrorFactories.grpcError(io.grpc.Status.RESOURCE_EXHAUSTED))
           }(DE)
           .flatMap(pollUntilPersisted(res._2, _))(DE)
     )
   }
 
+  private def prepareUploadPackageResult(
```

Comment:
This is not called anywhere.

Reply:
deleted this for now -- will add back any relevant similar parsing when polling is implemented